src/indexer.c

```diff
@@ -104,19 +106,24 @@ void *ofi_idx_remove(struct indexer *idx, int index)
 	union ofi_idx_entry *entry;
 	void *item;
 
+	if (!ofi_idx_is_valid(idx, index))
+		return NULL;
```

Comment:
I would drop this check from the remove call.  A caller can use lookup to validate an entry prior to removing it if it needs to knew that an invalid index might have been given.